src/updateManager.ts

```diff
@@ -30,28 +30,38 @@ export class UpdateManager {
 
     const succeeded = await this.db.transaction(async trx => {
       const rows = await trx(this.tableName)
-        .select('*')
-        .where('id', this.rowId)
-        .where('mv_locked', false)
-        // Only need to migrate if hash is different
-        .andWhereNot('mv_hash', newHash)
-        // And only if the previous hash is older than the current one
-        .andWhere('mv_ts', '<', isoTS).orWhereNull('mv_ts')
-        .forUpdate(); // This locks the row
-
-      // If row is found, lock it
+        .select(['id', 'mv_locked', 'mv_hash', 'mv_ts']).forUpdate()
+
+      // If rows is found, filter them
       if (rows.length) {
-        await trx(this.tableName).where('id', this.rowId).update({
-          mv_locked: true,
+        const filteredRows = rows.filter(row => {
+          return (
+              row.id === this.rowId &&
+              row.mv_locked === false &&
+              (row.mv_hash !== newHash || !row.mv_ts || row.mv_ts < isoTS)
+          );
         });
-        this._locked = {
-          hash: newHash,
-          ts: isoTS,
-        };
-        return true;
+        // If row is found, lock it
+        if (filteredRows.length){
+          await trx(this.tableName).where('id', this.rowId).update({
+            mv_locked: true,
+          });
+          this._locked = {
+            hash: newHash,
+            ts: isoTS,
+          };
+          return true;
+        }
+        return false
       }
-
-      return false;
+      // If no row is found (CI), create it
```

Comment:
The directus_setting row should be created by Directus, what was the scenario that lead you to needing to create a row?

Reply:
When initializing Directus for the first time with an empty database, the settings entry won't be created automatically. Instead, it will rely on the default values provided by Directus until the user explicitly saves the settings. I encountered this behavior using the example repository provided above, by debugging the extension and inspecting the database during initialization.

If the directus_settings entry doesn't exist, then this code
https://github.com/bcc-code/directus-schema-sync/blob/1065ee776f35f52f9f1ca1a1c16a72556404a62c/src/updateManager.ts#L27-L59
will return an empty `rows` object, resulting in a true return value for the `lockForUpdates()` method, so the import process won't happen.

https://github.com/bcc-code/directus-schema-sync/blob/1065ee776f35f52f9f1ca1a1c16a72556404a62c/src/index.ts#L89-L91